x/ccv/provider/keeper/relay.go

```diff
@@ -287,15 +294,19 @@ func (k Keeper) OnRecvSlashPacket(ctx sdk.Context, packet channeltypes.Packet, d
 func (k Keeper) ValidateSlashPacket(ctx sdk.Context, chainID string,
 	packet channeltypes.Packet, data ccv.SlashPacketData) error {
 
-	_, found := k.getMappedInfractionHeight(ctx, chainID, data.ValsetUpdateId)
 	// return error if we cannot find infraction height matching the validator update id
-	if !found {
-		return fmt.Errorf("cannot find infraction height matching "+
-			"the validator update id %d for chain %s", data.ValsetUpdateId, chainID)
+	if _, found := k.getMappedInfractionHeight(ctx, chainID, data.ValsetUpdateId); !found {
+		return providertypes.ErrSlashPacketInfractionHeightInvalid
```

Comment:
Error message can be improved or reverted to the previous approach. I'm unopinionated about how they look, it was easier (imho) to have errors defined in types so they can be checked more easily.